### ISSUE TITLE
[🐞] npm run build will not generate css in dist dir when i used unocss

### ISSUE THREAD

**Issue by cunzaizhuyi** [BuilderIO/qwik#2713]
### Which component is affected?

Qwik Runtime

### Describe the bug

I used unocss in my project, everything is ok in dev mode, but something wrong when i build;
anybody can help me?

i have tried the following things but not work:
* set sideEffects: ['*.css'] in package.json;
* reorder the `import 'uno.css'`'s location in file root.tsx;

### Reproduction

https://github.com/cunzaizhuyi/awesome-fe-sites

### Steps to reproduce

* git clone 
* npm run preview

### System Info

```shell
System:
    OS: macOS 13.0.1
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 2.74 GB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.9.0 - ~/.nvm/versions/node/v18.9.0/bin/node
    npm: 8.19.1 - ~/.nvm/versions/node/v18.9.0/bin/npm
  Browsers:
    Chrome: 109.0.5414.87
    Safari: 16.1
  npmPackages:
    @builder.io/qwik: 0.16.2 => 0.16.2 
    @builder.io/qwik-city: 0.0.128 => 0.0.128 
    undici: 5.14.0 => 5.14.0 
    vite: 4.0.3 => 4.0.3
```

**Comment by cunzaizhuyi** [comment]
I solved this problem by using unocss/runtime.
but thas is not a better way, for example, i dont know how to handle icons and how to config unocss.

**Comment by acoreyj** [comment]
ever have any luck with this?

**Comment by izcream** [comment]
got the same issues on my qwik-app too. Any solution update?

**Comment by yckbilly1929** [comment]
i'm trying to make unocss work with qwik in a monorepo (pnpm + nx)

i have put `import 'virtual:uno.css'` in root.tsx, but no css was generated except those imported preflights

**Comment by acoreyj** [comment]
You can always manually run the unocss cli as well

`npx unocss "./src/**" -o "./dist/uno.css" `

**Comment by acoreyj** [comment]
> 

can let me know what you added to the which config file?

**Comment by yckbilly1929** [comment]
> > i have put `import 'virtual:uno.css'` in root.tsx, but no css was generated except those imported preflights
> > @yckbilly1929
> 
> Can you let me know what you added to which config file

it's working as expected now, my setting is:
- unocss config in uno.config.ts
- Unocss() after qwikXXX plugins in vite.config.ts
- `import 'virtual:uno.css'` in src/root.tsx

**Comment by izcream** [comment]
> > > i have put `import 'virtual:uno.css'` in root.tsx, but no css was generated except those imported preflights
> > > @yckbilly1929
> > 
> > 
> > Can you let me know what you added to which config file
> 
> it's working as expected now, my setting is:
> 
> * unocss config in uno.config.ts
> * Unocss() after qwikXXX plugins in vite.config.ts
> * `import 'virtual:uno.css'` in src/root.tsx
> 
> for unocss-cli, i used it currently as last resort for esbuild-only SPA projects

is this methods work after build and serve with node server eg, express,fastify?

i try to move UnoCSS() after all qwik plugin in vite.config.ts but after i run serve it's still generate only preflights😅

**Comment by yckbilly1929** [comment]
yes, it works
i put it back before any qwikXXX plugins, seems order not related at all
one thing I did notice is `Unocss({ mode: 'disk-chunk' })` will break the build, and I have to switch back to default (global) mode for now

**Comment by izcream** [comment]
Its work on dev/preview mode or on production(build and server) too?

I try follow you way to setup everything work fine(dev/preview) but after I add fastify and build then run npm serve, website will not have any style from unocss except preflight

**Comment by yckbilly1929** [comment]
> Its work on dev/preview mode or on production(build and server) too?
> 
> I try follow you way to setup everything work fine(dev/preview) but after I add fastify and build then run npm serve, website will not have any style from unocss except preflight

Yes, it works on production build as well. I'm using fastify adapter.

You can make a quick check in server or dist folder after build. This is my full uno.config.ts setting:

```
import { FileSystemIconLoader } from '@iconify/utils/lib/loader/node-loaders'
import presetIcons from '@unocss/preset-icons'
import transformerCompileClass from '@unocss/transformer-compile-class'
import { defineConfig, presetUno } from 'unocss'

export default defineConfig({
  include: ['src/**/*.{html,tsx}'],
  exclude: ['node_modules', '.git', 'dist', 'server'],
  presets: [
    presetUno(),
    presetIcons({
      extraProperties: {
        display: 'inline-block',
        'vertical-align': 'middle',
      },
      collections: {
        images: FileSystemIconLoader('./src/assets/images'),
      },
    }),
  ],
  transformers: [transformerCompileClass()],
  preprocess(matcher) {
    return matcher.startsWith('un-') ? matcher.slice(3) : undefined // ignore
  },
})
```

**Comment by izcream** [comment]
Thanks I ll try again with this config😍

**Comment by izcream** [comment]
another workaround for make unocss-vite work with qwik is add `content.filesystem` in uno.config.ts(no idea about this config just lucky found😅)
```
export default defineConfig({
  content: {
    filesystem: ['src/**/*.{ts,tsx}']
  },
  presets: [presetUno(), presetIcons()],
  transformers: [transformerVariantGroup()]
})
```

**Comment by gooooooooney** [comment]
> 

hi, has this problem been resolved? I have encountered the same issue.

**Comment by gioboa** [comment]
@thejackshelton did you try UnoCSS recently? Is this issue still valid?
